src/OpenTelemetry.AutoInstrumentation/Configurations/Settings.cs

```diff
@@ -28,10 +28,10 @@ internal abstract class Settings
     /// </summary>
     public OtlpExportProtocol? OtlpExportProtocol { get; private set; }
 
-    public static T FromDefaultSources<T>()
+    public static T FromDefaultSources<T>(bool failFast)
         where T : Settings, new()
     {
-        var configuration = new Configuration(new EnvironmentConfigurationSource());
+        var configuration = new Configuration(failFast, new EnvironmentConfigurationSource(failFast));
```

Comment:
This seems like a chicken and egg problem, trying to embed a `failFast` value into the configuration source even though its value can be provided from a configuration source. I'm not requesting any changes, but this looks like an interesting circular dependency